src/commands/lightning/preview/app.ts

```diff
@@ -1,41 +1,137 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
 import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
-import { Messages } from '@salesforce/core';
+import { Messages, Logger, Org } from '@salesforce/core';
+import { PreviewUtils } from '@salesforce/lwc-dev-mobile-core';
+import { OrgUtils } from '../../../shared/orgUtils.js';
 
 Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
 const messages = Messages.loadMessages('@salesforce/plugin-lightning-dev', 'lightning.preview.app');
 
-export type LightningPreviewAppResult = {
-  path: string;
+enum Platform {
+  desktop = 'desktop',
+  ios = 'ios',
+  android = 'android',
+}
+
+type LightningPreviewAppFlags = {
+  name: string | undefined;
+  'target-org': Org;
+  'device-type': Platform;
+  'device-id': string | undefined;
 };
 
-export default class LightningPreviewApp extends SfCommand<LightningPreviewAppResult> {
+export default class LightningPreviewApp extends SfCommand<void> {
   public static readonly summary = messages.getMessage('summary');
   public static readonly description = messages.getMessage('description');
   public static readonly examples = messages.getMessages('examples');
 
   public static readonly flags = {
     name: Flags.string({
       summary: messages.getMessage('flags.name.summary'),
-      description: messages.getMessage('flags.name.description'),
       char: 'n',
-      required: false,
+    }),
+    'target-org': Flags.requiredOrg({
+      summary: messages.getMessage('flags.target-org.summary'),
+    }),
+    'device-type': Flags.option({
+      summary: messages.getMessage('flags.device-type.summary'),
+      char: 't',
+      options: [Platform.desktop, Platform.ios, Platform.android] as const,
+      default: Platform.desktop,
+    })(),
+    'device-id': Flags.string({
+      summary: messages.getMessage('flags.device-id.summary'),
+      char: 'i',
     }),
   };
 
-  public async run(): Promise<LightningPreviewAppResult> {
+  public async run(): Promise<void> {
     const { flags } = await this.parse(LightningPreviewApp);
+    const logger = await Logger.child(this.ctor.name);
+
+    if (flags['device-type'] === Platform.desktop) {
+      await this.desktopPreview(logger, flags);
+    } else if (flags['device-type'] === Platform.ios) {
+      await this.iosPreview(logger, flags);
+    } else if (flags['device-type'] === Platform.android) {
+      await this.androidPreview(logger, flags);
+    }
+  }
+
+  private async desktopPreview(logger: Logger, flags: LightningPreviewAppFlags): Promise<void> {
+    const appName = flags['name'];
+    const targetOrg = flags['target-org'];
+    const platform = flags['device-type'];
+
+    logger.debug('Determining Local Dev Server url');
+    // todo: figure out how to make the port dynamic instead of hard-coded value here
+    const ldpServerUrl = PreviewUtils.generateWebSocketUrlForLocalDevServer(platform, '8081');
```

Comment:
@nrkruk et al: I just want to call out that with this line we are (in this moment anyway) officially putting LDP-related utility functionality into the [lwc-dev-mobile-core](https://github.com/forcedotcom/lwc-dev-mobile-core) project/package. I think such functionality could live in there, or we could create a new independent package, or we could re-brand the existing one. No action necessary here: I just wanted to call it out for discussion.